batchflow/opensets/small_imagenet.py

```diff
@@ -0,0 +1,138 @@
+      Argument `drop_grayscale` controls whether grayscale images should be dropped.
+
+    """
+    SOURCE_URL = None
+    num_classes = 10
+
+    def __init__(self, *args, drop_grayscale=True, bar=False, preloaded=None, train_test=True, **kwargs):
+        self.bar = tqdm.tqdm(total=2) if  bar else None
+        self.drop_grayscale = drop_grayscale
+        super().__init__(*args, preloaded=preloaded, train_test=train_test, **kwargs)
+        if self.bar:
+            self.bar.close()
+
+    def download(self, path=None):
+        """ Load data from website and extract it into numpy arrays """
+
+        def _image_class(filepath):
+            """ Image's class is determined by the parent folder of the image """
+            return basename(dirname(filepath))
+
+        def _is_train(filepath):
+            """ Whether image belongs to train or val parts can be determined by
+            the level 2 parent folder of the image
+            """
+            return basename(dirname(dirname(filepath))) == 'train'
+
+        def _extract(archive_file, member):
+            data = archive_file.extractfile(member).read()
+            return PIL.Image.open(BytesIO(data))
+
+        def _gather_extracted(archive, files):
+            images = np.array([_extract(archive, file) for file in files], dtype=object)
+            labels = np.array([_image_class(file.name) for file in files])
+            labels_encoded = LabelEncoder().fit_transform(labels)
+
+            if self.drop_grayscale:
+                rgb_mask = [i for i in range(len(images)) if images[i].mode == 'RGB']
+                return images[rgb_mask], labels_encoded[rgb_mask]
+
+            return images, labels_encoded
+
+        if path is None:
+            path = tempfile.gettempdir()
+        filename = os.path.basename(self.SOURCE_URL)
+        localname = os.path.join(path, filename)
+
+        if not os.path.isfile(localname):
+            logger.info("Downloading %s", filename)
+            urllib.request.urlretrieve(self.SOURCE_URL, localname)
+            logger.info("Downloaded %s", filename)
+            if self.bar:
+                self.bar.update(1)
+
+        logger.info("Extracting...")
+        with tarfile.open(localname, "r:gz") as archive:
+            files_in_archive = archive.getmembers()
+
+            train_files = [file for file in files_in_archive if file.isfile() and _is_train(file.name)]
+            train_data = _gather_extracted(archive, train_files)
+
+            test_files = [file for file in files_in_archive if file.isfile() and not _is_train(file.name)]
+            test_data = _gather_extracted(archive, test_files)
+
+        logger.info("Extracted")
+        if self.bar:
+            self.bar.update(1)
+
+        self._train_index = DatasetIndex(np.arange(len(train_data[0])))
+        self._test_index = DatasetIndex(np.arange(len(test_data[0])))
+        return train_data, test_data
+
+
+class Imagenette(SmallImagenet):
```

Comment:
It seems that SmallImagenet is not needed at all. Just rename SmallImagenet to Imagenette and you'll reduce complexity of the code.